common-util/src/main/java/org/sunbird/services/sso/impl/KeyCloakServiceImpl.java

```diff
@@ -48,12 +48,14 @@
           + "realms/"
           + KeyCloakConnectionProvider.SSO_REALM
           + "/protocol/openid-connect/token";
-  private static final String SSO_PUBLIC_KEY = System.getenv(JsonKey.SSO_PUBLIC_KEY);
-
+  
+  private static final PublicKey SSO_PUBLIC_KEY = new KeyCloakRsaKeyFetcher()
+      .getPublicKeyFromKeyCloak(KeyCloakConnectionProvider.SSO_URL, KeyCloakConnectionProvider.SSO_REALM);
+  
   @Override
   public String verifyToken(String accessToken) {
     try {
-      PublicKey publicKey = toPublicKey(SSO_PUBLIC_KEY);
+      PublicKey publicKey = SSO_PUBLIC_KEY;
```

Comment:
We need to handle scenario when publicKey is null due to key cloak being temporarily down during lms service restart.

Reply:
we can throw exception if keycloak service is down  by checking if public key fetched is null .
Any other suggestions are welcomed